workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -494,19 +498,26 @@ def compute_first_rows_response(
                 use_auth_token=use_auth_token,
             )
             if not isinstance(iterable_dataset, IterableDataset):
-                raise TypeError("load_dataset should return an IterableDataset")
+                raise TypeError("load_dataset should return an IterableDataset. Please fix your loading script.")
             iterable_dataset = iterable_dataset._resolve_features()
             if not isinstance(iterable_dataset, IterableDataset):
-                raise TypeError("load_dataset should return an IterableDataset")
+                raise TypeError("load_dataset should return an IterableDataset. Please fix your loading script.")
             features = iterable_dataset.features
         except Exception as err:
-            raise FeaturesError("The split features (columns) cannot be extracted.", cause=err) from err
+            raise FeaturesError(
+                (
+                    f"Cannot extract the features (columns) for split '{split}' of config '{config}' of dataset"
+                    f" '{dataset}'."
+                ),
+                cause=err,
+            ) from err
     else:
         features = info.features
 
     if features and len(features) > columns_max_number:
         raise TooManyColumnsError(
-            f"Too many columns. The maximum supported number of columns is {columns_max_number}."
+            f"The number of columns ({len(features)}) exceeds the maximum supported number of columns"
+            f" ({columns_max_number})."
```

Comment:
Should we suggest to reduce the number of columns?

Reply:
Yes, good idea!

Generally, it comes from a flaw in the design of the dataset (not "tidy data"), but anyway, people have the right to do what they want.

So, I think we just want to say it's a limitation of the dataset viewer, and that by reducing the number of columns, it would be working again